internal/isoeditor/rhcos.go

```diff
@@ -6,66 +6,94 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
-	"sync"
+	"regexp"
 
-	"github.com/openshift/assisted-service/pkg/s3wrapper"
+	"github.com/openshift/assisted-service/internal/isoutil"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// CustomIsoTempDir is a temporary directory pattern for the custom ISO
+	CustomIsoTempDir string = "customiso"
+	// BaseIsoTempDir is a temporary directory pattern for the extracted base ISO
+	BaseIsoTempDir string = "baseiso"
+	// CustomIsoObjectName is a name pattern for the ISO template
+	CustomIsoObjectName string = "rhcos-%s-minimal.iso"
 )
 
 type Editor interface {
-	CreateMinimalISOTemplate(ctx context.Context, openshiftVersion string) error
+	CreateMinimalISOTemplate(ctx context.Context, openshiftVersion string) (string, error)
 }
 
 type rhcosEditor struct {
-	isoHandler    Handler
-	objectHandler s3wrapper.API
+	isoHandler    isoutil.Handler
 	rootfsURL     string
+	isoTmpWorkDir string
+	log           logrus.FieldLogger
 }
 
-func NewEditor(isoHandler Handler, objectHandler s3wrapper.API, rootfsURL string) Editor {
+func NewEditor(isoHandler isoutil.Handler, rootfsURL, isoTmpWorkDir string, log logrus.FieldLogger) Editor {
 	return &rhcosEditor{
-		isoHandler:    isoHandler,
-		objectHandler: objectHandler,
-		rootfsURL:     rootfsURL,
+		isoHandler: isoHandler,
+		rootfsURL:  rootfsURL,
+		log:        log,
 	}
 }
 
-func (e *rhcosEditor) CreateMinimalISOTemplate(ctx context.Context, openshiftVersion string) error {
+func CreateEditor(isoPath string, rootfsURL string, log logrus.FieldLogger) Editor {
+	isoTmpWorkDir, err := ioutil.TempDir("", BaseIsoTempDir)
+	if err != nil {
+		return nil
+	}
+	isoHandler := isoutil.NewHandler(isoPath, isoTmpWorkDir)
+	return NewEditor(isoHandler, rootfsURL, isoTmpWorkDir, log)
+}
+
+func GetMinimalIsoObject(openshiftVersion string) string {
+	return fmt.Sprintf(CustomIsoObjectName, openshiftVersion)
+}
+
+func GetRootfsURL(serviceBaseURL, openshiftVersion string) string {
+	return fmt.Sprintf("%s/api/assisted-install/v1/boot-files?file_type=rootfs.img&openshift_version=%s",
+		serviceBaseURL, openshiftVersion)
+}
```

Comment:
Is there something in our swagger library or maybe in the client code that stores this in a way that is more future proof?

Reply:
yes, used DownloadBootFilesURL.Build() instead